creator-node/src/ipfsClient.js

```diff
@@ -27,69 +25,115 @@ async function logIpfsPeerIds () {
+// TODO: Replace for images and figure out correct response structure
+/**
+ * Wrapper to ipfs.add() -- Allows enabling/disabling adding content to the ipfs daemon. Generally used for images (to generate dirs too)
+ * @param {Object} ipfs ipfs library -- can be two different versions (see ipfsClient.js)
+ * @param {Object[]} inputData an Object[] with the structure { path: string, content: buffer }
+ * @param {Object?} ipfsConfig ipfs add config options
+ * @param {Object?} logContext
+ * @param {boolean?} enableIPFSAdd flag to add content to ipfs daemon
+ * @returns {string|string[]|Object|Object[]} hashes from content addressing fn, or ipfs daemon responses
+ */
+// eslint-disable-next-line no-unused-vars
+async function ipfsMultipleAddWrapper (ipfs, inputData, ipfsConfig = {}, logContext = {}, enableIPFSAdd = false) {
+  const logger = genericLogger.child(logContext)
 
-  // If content was added to ipfs daemon, prioritize using that hash response
-  return ipfsDaemonHashes || onlyHashes
+  // Generate hashes with ipfs content hashing logic
+  const onlyHashes = await Promise.all(inputData.map(async ({ content }) => ipfsHashOf(content)))
+  const dirHash = await ipfsHashOf(inputData, {}, true /* isImgDir */)
+  onlyHashes.push(dirHash)
+
+  let ipfsDaemonResp
+  if (enableIPFSAdd) {
+    try {
+      ipfsDaemonResp = await ipfs.add(inputData, ipfsConfig)
+    } catch (e) {
+      logger.warn(`[ipfsClient - ipfsMultipleAddWrapper()] Could not add content to ipfs. Defaulting to onlyHash=${onlyHashes}: ${e.toString()}`)
+      return onlyHashes
+    }
+  }
+
+  // Return the `ipfsDaemonHashes`, or `onlyHashes`. Prioritize `ipfsDaemonHashes`.
+  if (ipfsDaemonResp) {
+    const ipfsDaemonHashes = ipfsDaemonResp.map(entry => entry.hash)
+    logger.info(`[ipfsClient - ipfsMultipleAddWrapper()] onlyHash=${onlyHashes} ipfsDaemonHash=${ipfsDaemonHashes.toString()} isSameHash=${onlyHashes.toString() === ipfsDaemonHashes.toString()}`)
+    return ipfsDaemonResp
+  } else {
+    logger.info(`[ipfsClient - ipfsMultipleAddWrapper()] onlyHash=${onlyHashes}`)
+    return onlyHashes
+  }
 }
 
 /**
- * Wrapper to ipfs.add() -- This wrapper allows for ipfs calls to be made async or synchronous.
+ * Wrapper to ipfs.addFromFs() -- Allows enabling/disabling adding content to the ipfs daemon.
  * @param {Object} ipfs ipfs library -- can be two different versions (see ipfsClient.js)
  * @param {string|string[]} srcPath
  * @param {Object?} ipfsConfig ipfs add config options
  * @param {Object?} logContext
  * @returns {string|string[]|Object|Object[]} hashes from content addressing fn, or ipfs daemon responses
  */
-async function ipfsAddFromFsWrapper (ipfs, srcPath, ipfsConfig = {}, logContext = {}) {
+async function ipfsAddFromFsWrapper (ipfs, srcPath, ipfsConfig = {}, logContext = {}, enableIPFSAdd = false) {
   const logger = genericLogger.child(logContext)
 
+  // Generate hash with ipfs content hashing logic
   const stream = fs.createReadStream(srcPath)
```

Comment:
is it possible to call ipfsSingleAddWrapper with this stream instead of redefining this function?